src/jni_core/src/lib.rs

```diff
@@ -296,30 +298,44 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_vnodeCount(
     VirtualNode::COUNT as jint
 }
 
-#[cfg(not(madsim))]
 #[no_mangle]
 pub extern "system" fn Java_com_risingwave_java_binding_Binding_hummockIteratorNew<'a>(
     env: EnvParam<'a>,
     read_plan: JByteArray<'a>,
 ) -> Pointer<'static, HummockJavaBindingIterator> {
     execute_and_catch(env, move |env| {
-        let read_plan = Message::decode(to_guarded_slice(&read_plan, env)?.deref())?;
-        let iter = RUNTIME.block_on(HummockJavaBindingIterator::new(read_plan))?;
-        Ok(iter.into())
+        #[cfg(madsim)]
```

Comment:
Does [`#[cfg_or_panic(not(madsim))]`](https://docs.rs/cfg-or-panic/0.2.0/cfg_or_panic/attr.cfg_or_panic.html#) work here?

Reply:
Great to know that there is such a useful macro. Thanks for the suggestion!